api/client.go

```diff
@@ -373,18 +405,20 @@ func NewClient(c *Config) (*Client, error) {
 		return nil, errwrap.Wrapf("error encountered setting up default configuration: {{err}}", def.Error)
 	}
 
+	var clientConfig *Config
 	if c == nil {
-		c = def
+		clientConfig = def
+	} else {
+		clientConfig = c.Clone()
 	}
 
 	c.modifyLock.Lock()
```

Comment:
We shouldn't need this lock if we're not touching c any longer, and since we have a clone we shouldn't need to touch c any longer.  Maybe assign c=nil to enforce that?